crates/swc_ecma_minifier/tests/fixture/issues/6636/output.js

```diff
@@ -0,0 +1,8 @@
+export function memo(getDeps, fn, opts) {
```

Comment:
Previous output:

```js

export function memo(getDeps, fn, opts) {
    let result, deps = [];
    return ()=>{
        opts.key && opts.debug && Date.now();
        const newDeps = getDeps(), depsChanged = newDeps.length !== deps.length || newDeps.some((dep, index)=>deps[index] !== dep);
        return depsChanged && (deps = newDeps, opts.key && opts.debug && Date.now(), opts?.onChange?.(result = fn(...newDeps)), opts.key && opts.debug && opts?.debug()), result;
    };
}
```

Reply:
Can we reduce this down further? Seems like https://play.swc.rs/?version=1.3.22&code=H4sIAAAAAAAAA0XMQQqDMBCF4b2neMsEylwgWBftRYpMbEFnQjLBQvHuVhvo8vE9fn4nzYZYZbSXChZe1EW5QJMVj08HzGzIXOps4bsyW80C59FfT0ZD9IjiiEh4vXMqPpx4dAZSuT0fMvFA7vdu2mr%2F%2FBa6bQf5UtPElAAAAA%3D%3D&config=H4sIAAAAAAAAA3VUQZLjIAz8C%2Bc5ZOc0NQ%2FY27yBIiAcshi5kMjElcrfVyQ2xMnMzWpaQlK3uagjWfV5UZPJBLl%2B0ZzYnNWn4nkCsjlMrN4Uk0DeRIKrBCYPwEIBet%2B97%2BQ4IhIshDc1hhT8XItZHKcMRPVbksoIianxTM74LSHnItEeMYJJL7E2pENiGCD3VIsxmolAn0xuGfU2kwNh2kCFwekp4%2FSAJhc4YJJiK%2BbAOG3RQQNCBsvhBJ0iNYSSSBprndxAB%2FsyDHWBdyacTCyGWy0439YgN7a8AwZi7UuiJ2jT6R26D7mwgtcZuOS0co4Y0mYP%2FwCkz2iIkhmhZ95wLzpsmf4HVkheROS5oaJw7yrBIMPpEHybsHYNmUPfVgZXLNQJbS%2B%2BgJuBKDjQ4L1suyXTd2B76MWqGdG3ULZmfNfgHurVT1u0WuAF%2FCtd8aMcCz4aPjxjNI97jC8lRuADuhdYFsH4DGZx73l6RktyIGsG93BQqIbNpiI%2Bo471b1v3LIaQHD1E3Df7yk85mjRUW14U4xThBHGr6K9%2B%2BAGTS05V4MWKq%2FHgo2tmxEfhz259FK61A3Rl6UDUkudBNj9iOpK69jdhrUVfC3vRG0u28GXqdSqkKBOq63%2Fb7HJPngQAAA%3D%3D can repro the issue.